index.bs

```diff
@@ -5474,14 +5494,213 @@ A [=remote end=] has a <dfn>navigable cache behavior map</dfn> which is a weak
+
+Note: This hook is intended to be triggered by the fetch spec when the response is set.
+
+1. For each |session| in [=active BiDi sessions=]:
+
+   Note: Should check for collectors matching navigable + response here already?
+   Otherwise we can do it only in [=maybe collect network response body=] so that
+   clients can add the collector as late as possible?
+
+   1. If |session|'s [=network collectors=] is not [=list/empty=]:
+
+      1. Let |collected body| be a struct with
+            <code>bytes</code> field set to null,
+            <code>cloned body</code> field set to [=body/clone=] with |response body|,
+            <code>collectors</code> field set to an empty list,
+            <code>request</code> field set to |request|'s [=request id=],
+            <code>size</code> field set to null,
+            <code>type</code> field set to "response".
+
+      1. [=list/Append=] |collected body| to [=collected network bodies=].
+
+      1. Return.
+
+</div>
+
+<div algorithm>
+To <dfn>get collected body</dfn> given |request id| and |body type|.
+
+1. For |collected body| of [=collected network bodies=]:
+
+   1. If |collected body|'s <code>request</code> is |request id| and
+      |collected body|'s <code>type</code> is |body type|, return |collected body|.
+
+1. Return null.
+
+</div>
+
+<div algorithm>
+To <dfn>maybe collect network response body</dfn> given |session| and |request|:
+
+1. Let |navigable| be [=get navigable for request=] with |request|.
+
+1. If |navigable| is null, return.
+
+NOTE: This prevents from collecting bodies not related to a navigable.
+We still need to retrieve the navigable to check against the collector
+configuration but we could still accept null here.
+
+1. Let |top-level navigable| be |navigable|'s [=navigable/top-level traversable=].
+
+1. Let |collectors| be an empty list.
+
+1. For each |collector| in |session|'s [=network collectors=]:
+
+   1. If |collector|'s <code>body types</code> [=list/contains=] "response" and if [=match collector for navigable=] with |collector| and |top-level navigable|:
+
+      1. [=list/Append=] |collector| to |collectors|.
+
+1. If |collectors| is [=list/empty=], return.
+
+1. Let |collected body| be [=get collected body=] with |request|'s [=request id=] and "response".
+
+1. If |collected body| is null, return.
+
+Note: This might happen if there are no collectors setup when the response is created,
+and [=clone network response body=] does not clone the corresponding body.
+
+1. Let |bytes| be null.
+
+1. Let |size| be null.
+
+1. Let |processBody| given |nullOrBytes| be this step:
+
+   1. If |nullOrBytes| is not null:
+
+      1. Set |bytes| to [=serialize protocol bytes=] with |nullOrBytes|.
+
+      1. Set |size| to |nullOrBytes|'s length.
+
+1. Let |processBodyError| be this step: Do nothing.
+
+1. [=Fully read=] |collected body|’s <code>cloned body</code> given |processBody| and |processBodyError|.
```

Comment:
To clarify: the body would be the raw network body before i.e. content decoding? or is the fetch spec doing the decoding for us?

Reply:
Good question.

Based on https://fetch.spec.whatwg.org/#http-network-fetch

Step 16.1.1

The incoming bytes seem to be decoded on the fly via https://fetch.spec.whatwg.org/#handle-content-codings and added to `buffer` which is used by the `pullAlgorithm` for the `response body`'s `stream`. So I think fetch is doing the decoding for us here, and reading the stream should directly give access to decoded content?

Reply:
I wonder if that is what we want for the use cases such as HAR generation? Would we have a content encoding set to something like gzip and the body to actually not be gzipped anymore?

Reply:
I don't think this is an issue for har. Gzipped content that can be decoded as text is usually just saved as decoded text in HAR files. And other resources such as images will typically be base64 encoded. 

See HAR "spec":
> Before setting the text field, the HTTP response is decoded (decompressed & unchunked), than trans-coded from its original character set into UTF-8. Additionally, it can be encoded using e.g. base64. Ideally, the application should be able to unencode a base64 blob and get a byte-for-byte identical resource to what the browser operated on.
> 
> Encoding field is useful for including binary responses (e.g. images) into the HAR file.